### PR TITLE
Docs/release notes 0.36.0

### PR DESCRIPTION
**Pull request by rhero**
**Goals (and why)**: Cleaning up 0.36.0 release notes so I can email them out.

**Implementation Description (bullets)**:

- made 0.36.0 table
- cleaned up wording for notes in current release
- moved a few release notes that were in the wrong section. Here are the changes in this release for reference.

```
rhero$ git log --pretty=format:"%h%x09%an%x09%ad%x09%s" --date=short 0.35.0...0.36.0 | grep "#"
5462f1b	Glenn Sheasby	2017-03-15	Merge pull request #1703 from palantir/fix/dbkvs-test-locks
b0a85ec	Jeremy Kong	2017-03-14	Docs Improvements (largely around TimeLock) (#1700)
9bd97bf	Greg Bonik	2017-03-13	Add benchmarks for getRowsColumnRange() over a single wide row (#1684)
e772f8c	Glenn Sheasby	2017-03-10	Merge pull request #1696 from palantir/fix/hide-persisted-locks-table
643a5e1	Glenn Sheasby	2017-03-10	Merge pull request #1678 from palantir/fix/dbkvs-oom
54baa02	Himangi Saraogi	2017-03-10	Merge pull request #1685 from palantir/ds/tritium
8239e7f	Jeremy Kong	2017-03-09	Declare TransactionManager.close() explicitly (so it will throw no checked exceptions) (#1677)
05371eb	Glenn Sheasby	2017-03-08	Merge pull request #1681 from palantir/devex/paxos-leader-election-service
6201a71	Robert Hero	2017-03-08	Merge pull request #1692 from palantir/rhero-patch-1
e422c8e	Glenn Sheasby	2017-03-08	Merge pull request #1682 from palantir/fix/sweep-stats-count-delete-range-all
255b38f	Robert Hero	2017-03-08	Merge pull request #1691 from palantir/docs/release-notes-0.35.0
95f9eab	Robert Hero	2017-03-08	remove getting started notes and replaced with actual startup command (#1663)
314c46c	Glenn Sheasby	2017-03-07	Merge pull request #1521 from palantir/fix/keep-read-only-safe
74e4360	Jeremy Kong	2017-03-07	ETE Tests of TimeLock Migration [TimeLock ETE Tests, Part III] (#1676)
26610d8	Glenn Sheasby	2017-03-07	Merge pull request #1687 from palantir/fix_crappy_logging
4df45d2	Robert Hero	2017-03-06	Merge pull request #1662 from palantir/docs/update-tx-protocol
f39387e	Glenn Sheasby	2017-03-06	Merge pull request #1489 from palantir/add/publish-protos-used-internally
3735163	John Boreiko	2017-03-06	Merge pull request #1593 from palantir/fix/prefixed-table-names
```

**Concerns (what feedback would you like?)**: Make sure spelling and wording make sense. Would like a response to specific comments as well. 

**Where should we start reviewing?**: release-notes.rst

**Priority (whenever / two weeks / yesterday)**: asap please!

### CODE REVIEW

File: docs/source/release_notes/release-notes.rst [hunk 1]
```diff
@@ -40,8 +40,29 @@ develop
...
+    :widths: 5 40
+    :header-rows: 1
+
+    *    - Type
+         - Change
+
     *    - |fixed|
-         - Fixed DbKvs sweep OOM issue (`#982 <https://github.com/palantir/atlasdb/issues/982>`__) caused by very wide rows. ``DbKvs.getRangeOfTimestamps`` now uses an adjustable cell batch size to avoid loading too many timestamps. In case of a single row that is too wide, this may result in ``getRangeOfTimestamps`` returning multiple ``RowResult`` to include all timestamps. It is, however, guaranteed that each ``RowResult`` will contain all timestamps for each included column.
+         - Fixed DBKVS sweep OOM issue (`#982 <https://github.com/palantir/atlasdb/issues/982>`__) caused by very wide rows.
+           ``DbKvs.getRangeOfTimestamps`` now uses an adjustable cell batch size to avoid loading too many timestamps.
```
Comment (rhero) [inline docs/source/release_notes/release-notes.rst]: For the "uses an adjustable cell batch size..." part, does that mean it's a user configurable option? We should probably call out the actual parameter, or link to documentation that describes it.

Comment (rhero) [inline docs/source/release_notes/release-notes.rst]: @gmaretic ^

File: docs/source/release_notes/release-notes.rst [hunk 2]
```diff
@@ -40,8 +40,29 @@ develop
...
+           In case of a single row that is too wide, this may result in ``getRangeOfTimestamps`` returning multiple ``RowResult`` to include all timestamps.
+           It is, however, guaranteed that each ``RowResult`` will contain all timestamps for each included column.
            (`Pull Request <https://github.com/palantir/atlasdb/pull/1678>`__)
 
     *    - |fixed|
```
Comment (rhero) [inline docs/source/release_notes/release-notes.rst]: @jboreiko Can you describe what the old failure scenario looked like? We say, "we allowed some incorrect behavior before, and now we prevent that behavior", but what are the practical ramifications of allowing the old behavior?

File: docs/source/release_notes/release-notes.rst [hunk 3]
```diff
@@ -57,13 +78,33 @@ develop
            (`Pull Request <https://github.com/palantir/atlasdb/pull/1687>`__)
 
     *    - |fixed|
-         - Fixed an issue where the ``_persisted_locks`` table was erroneously reported not to have persisted metadata.
+         - Fixed an issue where the ``_persisted_locks`` table was unnecessarily logged as not have persisted metadata.
```
Comment (gsheasby) [inline docs/source/release_notes/release-notes.rst]: nit: should be "as not having"

File: docs/source/release_notes/release-notes.rst (hunk 1 above)
Comment (gsheasby) [inline docs/source/release_notes/release-notes.rst]: The batch size can, at present, be adjusted by calling `DbKvs.setMaxRangeOfTimestampsBatchSize`, but I believe it should be controlled via AtlasDB config.

So for this release, we might want to say that we use a cell batch size of 1 million, and then provide a config option in time for the next release.

Comment (rhero) [inline docs/source/release_notes/release-notes.rst]: So it can be set by products, but it can't be set by users, and our default is 1 million?

Comment (gmaretic) [inline docs/source/release_notes/release-notes.rst]: Yes

File: docs/source/release_notes/release-notes.rst (hunk 2 above)
Comment (gmaretic) [inline docs/source/release_notes/release-notes.rst]: The old behaviour was that we would try to load all timestamps from the requested range into memory, and if there were too many of them we would OOM. Now we make sure to load only as much as 1M + all timestamps for one cell instead. 
From the point of view of the caller of the method, nothing changed except in the case where a single row is so wide (more than 1M timestamps in that row) that it has to be split into several row results.

**Pull request merged**